extension/encoding/awslogsencodingextension/internal/unmarshaler/cloudtraillogs/cloudtraillogs.go

```diff
@@ -0,0 +1,290 @@
+	Resources                    []any          `json:"resources"`
+	ReadOnly                     bool           `json:"readOnly"`
+	ManagementEvent              bool           `json:"managementEvent"`
+	AdditionalEventData          map[string]any `json:"additionalEventData"`
+	TLSDetails                   map[string]any `json:"tlsDetails"`
+	SessionCredentialFromConsole string         `json:"sessionCredentialFromConsole"`
+}
+
+type CloudTrailLogs struct {
+	Records []CloudTrailRecord `json:"Records"`
+}
+
+func NewCloudTrailLogsUnmarshaler(buildInfo component.BuildInfo) plog.Unmarshaler {
+	return &CloudTrailLogsUnmarshaler{
+		buildInfo: buildInfo,
+		gzipPool:  sync.Pool{},
+	}
+}
+
+func (u *CloudTrailLogsUnmarshaler) Unmarshal(buf []byte) (plog.Logs, error) {
+	return u.UnmarshalLogs(buf)
+}
+
+func (u *CloudTrailLogsUnmarshaler) UnmarshalLogs(buf []byte) (plog.Logs, error) {
+	gzipReader, ok := u.gzipPool.Get().(*gzip.Reader)
+	if !ok {
+		var err error
+		gzipReader, err = gzip.NewReader(bytes.NewReader(buf))
+		if err != nil {
+			return plog.Logs{}, fmt.Errorf("failed to create gzip reader: %w", err)
+		}
+	} else {
+		if err := gzipReader.Reset(bytes.NewReader(buf)); err != nil {
+			u.gzipPool.Put(gzipReader)
+			return plog.Logs{}, fmt.Errorf("failed to reset gzip reader: %w", err)
+		}
+	}
+
+	defer func() {
+		_ = gzipReader.Close()
+		u.gzipPool.Put(gzipReader)
+	}()
+
+	decompressedBuf, err := io.ReadAll(gzipReader)
+	if err != nil {
+		return plog.Logs{}, fmt.Errorf("failed to decompress CloudTrail logs: %w", err)
+	}
+
+	var cloudTrailLogs CloudTrailLogs
+	if err := json.Unmarshal(decompressedBuf, &cloudTrailLogs); err != nil || len(cloudTrailLogs.Records) == 0 {
+		return plog.Logs{}, fmt.Errorf("failed to unmarshal CloudTrail logs: %w", err)
+	}
+
+	return u.processRecords(cloudTrailLogs.Records), nil
+}
+
+func (u *CloudTrailLogsUnmarshaler) processRecords(records []CloudTrailRecord) plog.Logs {
+	logs := plog.NewLogs()
+
+	if len(records) == 0 {
+		return logs
+	}
+
+	// Create a single resource logs entry for all records
+	resourceLogs := logs.ResourceLogs().AppendEmpty()
+	scopeLogs := resourceLogs.ScopeLogs().AppendEmpty()
+	scopeLogs.Scope().SetName(metadata.ScopeName)
+	scopeLogs.Scope().SetVersion(u.buildInfo.Version)
+
+	// Set resource attributes based on the first record
+	// (all records have the same account ID and region)
+	u.setResourceAttributes(resourceLogs.Resource().Attributes(), records[0])
+
+	for _, record := range records {
+		logRecord := scopeLogs.LogRecords().AppendEmpty()
+		u.setLogRecord(logRecord, record)
+	}
+
+	return logs
+}
+
+// checks if the event is related to EC2 instance operations
+func isEC2InstanceOperation(eventName string) bool {
+	ec2Operations := map[string]bool{
+		"StartInstances":     true,
+		"StopInstances":      true,
+		"TerminateInstances": true,
+		"RebootInstances":    true,
+		"RunInstances":       true,
+	}
+	return ec2Operations[eventName]
+}
```

Comment:
Not to be a pain, but maybe just use `switch` instead of a map lookup? :)  It is better for performance, but I doubt it would should on an overall benchmark